akka-cluster/src/main/scala/akka/cluster/ClusterEvent.scala

```diff
@@ -268,13 +268,12 @@ object ClusterEvent {
   private[cluster] def diffUnreachable(oldState: MembershipState, newState: MembershipState): immutable.Seq[UnreachableMember] =
     if (newState eq oldState) Nil
     else {
-      val oldGossip = oldState.latestGossip
       val newGossip = newState.latestGossip
-      val oldUnreachableNodes = oldGossip.overview.reachability.allUnreachableOrTerminated
-      (newGossip.overview.reachability.allUnreachableOrTerminated.collect {
+      val oldUnreachableNodes = oldState.dcReachabilityNoOutsideNodes.allUnreachableOrTerminated
+      newState.dcReachabilityNoOutsideNodes.allUnreachableOrTerminated.collect {
         case node if !oldUnreachableNodes.contains(node) && node != newState.selfUniqueAddress ⇒
           UnreachableMember(newGossip.member(node))
-      })(collection.breakOut)
+      }(collection.breakOut)
     }
```

Comment:
perhaps add some test in `ClusterDomainEventSpec` to verify that events are not emitted for some cross cases (it probably already covers the "within" cases)